src/ApiService/ApiService/onefuzzlib/orm/Orm.cs

```diff
@@ -0,0 +1,265 @@
+{
+    private readonly JsonSerializerOptions _options;
+
+    private readonly ConcurrentDictionary<Type, EntityInfo> _cache;
+
+
+    public EntityConverter()
+    {
+        _options = new JsonSerializerOptions()
+        {
+            PropertyNamingPolicy = new OnefuzzNamingPolicy(),
+        };
+        _options.Converters.Add(new CustomEnumConverterFactory());
+        _cache = new ConcurrentDictionary<Type, EntityInfo>();
+    }
+
+    internal Func<object[], object> BuildConstructerFrom(ConstructorInfo constructorInfo)
+    {
+        var constructorParameters = Expression.Parameter(typeof(object?[]));
+
+        var parameterExpressions =
+            constructorInfo.GetParameters().Select((parameterInfo, i) =>
+            {
+                var ithIndex = Expression.Constant(i);
+                var ithParameter = Expression.ArrayIndex(constructorParameters, ithIndex);
+                var unboxedIthParameter = Expression.Convert(ithParameter, parameterInfo.ParameterType);
+                return unboxedIthParameter;
+
+            }).ToArray();
+
+        NewExpression constructorCall = Expression.New(constructorInfo, parameterExpressions);
+
+        Func<object[], object> ctor = Expression.Lambda<Func<object[], object>>(constructorCall, constructorParameters).Compile();
+        return ctor;
+    }
+
+    private EntityInfo GetEntityInfo<T>()
+    {
+        return _cache.GetOrAdd(typeof(T), type =>
+        {
+            var constructor = type.GetConstructors()[0];
+            var parameterInfos = constructor.GetParameters();
+            var parameters =
+            parameterInfos.Select(f =>
+            {
+                var isRowkey = f.GetCustomAttribute(typeof(RowKeyAttribute)) != null;
+                var isPartitionkey = f.GetCustomAttribute(typeof(PartitionKeyAttribute)) != null;
+
+
+                var (dbName, kind) =
+                isRowkey
+                    ? ("RowKey", EntityPropertyKind.RowKey)
+                    : isPartitionkey
+                        ? ("PartitionKey", EntityPropertyKind.PartitionKey)
+                        : (// JsonPropertyNameAttribute can only be applied to properties
+                            typeof(T).GetProperty(f.Name)?.GetCustomAttribute<JsonPropertyNameAttribute>()?.Name
+                                ?? CaseConverter.PascalToSnake(f.Name),
+                            EntityPropertyKind.Column
+                        );
+                if (f.Name == null)
+                {
+                    throw new Exception();
+                }
+
+                if (f.ParameterType == null)
+                {
+                    throw new Exception();
+                }
+
+                return new EntityProperty(f.Name, dbName, f.ParameterType, kind);
+            }).ToArray();
+
+            return new EntityInfo(typeof(T), parameters, BuildConstructerFrom(constructor));
+        });
+    }
+
+    public TableEntity ToTableEntity<T>(T typedEntity)
```

Comment:
https://github.com/stishkin/onefuzz/blob/44da180e2646d1f10042cfd345b2d45c30e0d98b/src/api-service/__app__/onefuzzlib/orm.py#L301

Python Models have some "filters" specified to make sure that some fields are not stored in the tables. 

Is there plan to add that in next pr ?  Since that will require handling of Ignore attribute (not just Name attrtibute)

Reply:
no needed in our case here. The code looks at the parameter of the record constructor and maps it to the db. 
`excludes` where used for data that was added after the fact. for those we can use the property syntax of the record.
``` C#
/// the record constructor parameters will save in the db
record MyEntity([PartitonKey] string name, [RowKey] int id)
{
        /// this property will not be saved in the db
	public string AdditionalProperty {get; set;}
}
```

Reply:
so that's a convention we need to remember